build/nodes-and-endpoints/join-mainnet.md

```diff
@@ -336,22 +336,61 @@ git checkout $BINARY_VERSION
 make build
 cp build/persistenceCore ~/.persistenceCore/cosmovisor/upgrades/$BINARY_VERSION/bin
 ```
-
 Now, at the upgrade height, **Cosmovisor** will upgrade swap the binaries.
 
-### Manual Option
+**2. Auto-download option**   
+Above manual option, still needs upgrade version binary to be manually added under cosmovisor/upgrades/upgrade-name/bin directory.
+For clarity, refer below folder layout for daemon home directory:-
+
+![alt text for screen readers](cosmo-visor-folder-layout.jpg)
+
+However, for people who don't need such control and want an automated setup to download binaries on all the validating nodes, there is an **auto-download** option.
+
+**NOTE**: We don't recommend using auto-download because it doesn't verify in advance if a binary is available. If there will be any issue with downloading a binary, the cosmovisor will stop and won't restart an App (which could lead to a chain halt). Therefore, it is every node operator's responsibility to analyze such risks and use cosmovisor as per requirement.
+
+#### Steps to enable the Auto-Download feature
+**Case A: When creating the governance proposal (skip if you're a node operator)**
+1. Create a Github release for target version binary/tar for all required environments and calculate the checksum using sha256sum or sha512sum. The downloadable binary path with checksum can be drafted then as below:
+    ```   
+    "https://github.com/cosmos/gaia/releases/download/v8.0.0/gaiad-v8.0.0-linux-amd64?checksum=sha256:6d0c123e246a8b56ba534f70dd5dc72058b00fd5e5dde5ea40509ff51efc42e2"
+    ```
+2. Create a JSON file in format:- os/architecture -> binary URI map under the "binaries" key. Note that we can list multiple binaries for appropriate environments in this file.  
+   For example:
+    ```json
+    {
+      "binaries": {
+        "linux/amd64": "https://github.com/cosmos/gaia/releases/download/v8.0.0/gaiad-v8.0.0-linux-amd64?checksum=sha256:6d0c123e246a8b56ba534f70dd5dc72058b00fd5e5dde5ea40509ff51efc42e2",
```

Comment:
Can we use a persistence example instead of gaia?

Reply:
noted